install/kubernetes/cilium/values.yaml

```diff
@@ -443,6 +443,12 @@ encryption:
     # -- The interface to use for encrypted traffic.
     interface: ""
 
+    # -- Whether to create the secret.
+    createSecret: false
+
+    # -- Secret key data to use when creating the secret. e.g., "3 rfc4106(gcm(aes)) c971f5e7a633b2e5ff248b53995784026fd48873 128".
+    keyData: ""
```

Comment:
probably questions to other team member:

- how confidential is this key ? 
- with this PR, this key value can be managed internally, what happens if the value is changed between installations/upgrades/etc ?

Reply:
> how confidential is this key

I'm not sure, but since helm keeps releases in secrets, and this is already stored as a secret when managed externally, might that be equivalent?

> what happens if the value is changed between installations/upgrades/etc

From the docs, it looks like key rotation needs cilium agents to restart: https://github.com/cilium/cilium/blob/ed14af8b4c514deef12dcd1a12678de5110352b1/install/kubernetes/cilium/templates/cilium-configmap.yaml#L444-L448

Looks like there's already a hash of the config label in the agent daemonset, we could add a similar hash of this secret?

Reply:
> Looks like there's already a hash of the config label in the agent daemonset, we could add a similar hash of this secret?

This is exactly what I am heading next, we might need to roll out automatically based on rollOut* attributes. I didn't have time to test it out if there is any implication if we are doing so (in case there is change in value).

Reply:
Want to generalize it already then, and create a helper for agent rollOut* attributes? Thinking something like:

```
{{- define "cilium.agent.rollOutConfigHash" -}}
{{- $items := list .Values.encryption.ipsec.keyData -}}
{{- printf "%s" (join $items "") | sha256sum -}}
{{- end -}}
```

Reply:
I think we can reuse the same attribute rollOutCiliumPods, which is an indicator to add secret hash to pod template annotation.

Currently, I am not sure how to generalize for different scenarios (e.g. enabled vs disabled encryption, self managed secret vs external secret, etc). Feel free to share your implementation here, thanks.

https://github.com/cilium/cilium/blob/6c1e09ad3e29bd8dcfeb805ef7fb900ebbf5f138/install/kubernetes/cilium/values.yaml#L80

https://github.com/cilium/cilium/blob/6f0a866366f2420383649c1dc048a1b1844c49ac/install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml#L42